`getKubernetesVersion()` fails with Kubernetes 1.33

### Describe the bug

When running the `KuebrnetesClient.getKubernetesVersion()` method against a Kube 1.33 cluster, it fails to decode the information due to new fields being provided.

```
Exception in thread "main" io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.copyAsCause(KubernetesClientException.java:205)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.waitForResult(OperationSupport.java:507)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handleResponse(OperationSupport.java:524)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.restCall(OperationSupport.java:693)
	at io.fabric8.kubernetes.client.impl.BaseClient.getVersionInfo(BaseClient.java:311)
	at io.fabric8.kubernetes.client.impl.KubernetesClientImpl.getKubernetesVersion(KubernetesClientImpl.java:632)
	at cz.scholz.sandbox.fabric8.VersionInfo.main(VersionInfo.java:16)
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:109)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:102)
	at io.fabric8.kubernetes.client.utils.KubernetesSerialization.unmarshal(KubernetesSerialization.java:276)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.lambda$handleResponse$0(OperationSupport.java:551)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$completeOrCancel$10(StandardHttpClient.java:141)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at io.fabric8.kubernetes.client.http.ByteArrayBodyHandler.onBodyDone(ByteArrayBodyHandler.java:51)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at io.fabric8.kubernetes.client.vertx.VertxHttpRequest.lambda$consumeBytes$1(VertxHttpRequest.java:84)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:270)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:252)
	at io.vertx.core.http.impl.HttpEventHandler.handleEnd(HttpEventHandler.java:76)
	at io.vertx.core.http.impl.HttpClientResponseImpl.handleEnd(HttpClientResponseImpl.java:250)
	at io.vertx.core.http.impl.Http1xClientConnection$StreamImpl.lambda$new$0(Http1xClientConnection.java:424)
	at io.vertx.core.streams.impl.InboundBuffer.handleEvent(InboundBuffer.java:279)
	at io.vertx.core.streams.impl.InboundBuffer.write(InboundBuffer.java:157)
	at io.vertx.core.http.impl.Http1xClientConnection$StreamImpl.handleEnd(Http1xClientConnection.java:734)
	at io.vertx.core.impl.ContextImpl.execute(ContextImpl.java:327)
	at io.vertx.core.impl.ContextImpl.execute(ContextImpl.java:307)
	at io.vertx.core.http.impl.Http1xClientConnection.handleResponseEnd(Http1xClientConnection.java:965)
	at io.vertx.core.http.impl.Http1xClientConnection.handleHttpMessage(Http1xClientConnection.java:837)
	at io.vertx.core.http.impl.Http1xClientConnection.handleMessage(Http1xClientConnection.java:799)
	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:159)
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1515)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1378)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1427)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:530)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:469)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:796)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "emulationMajor" (class io.fabric8.kubernetes.client.VersionInfo), not marked as ignorable (9 known properties: "goVersion", "gitTreeState", "platform", "minor", "gitVersion", "gitCommit", "buildDate", "compiler", "major"])
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 4, column: 22] (through reference chain: io.fabric8.kubernetes.client.VersionInfo["emulationMajor"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1153)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:2224)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1719)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1697)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:316)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2125)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1501)
	at io.fabric8.kubernetes.client.utils.KubernetesSerialization.unmarshal(KubernetesSerialization.java:272)
	... 62 more
```

This seems to be caused by the new fields in the `/version` endpoint.

I can open a PR with the new fields in the `VersionInfo` file. But I'm not sure if that is the right solution and whether the file isn't generated from some spec etc.

### Fabric8 Kubernetes Client version

7.1.0

### Steps to reproduce

1. Deploy Kube 1.33 cluster (in my case a simple cluster based on the Kubeadm tool)
2. Run the following Java code based on Fabric8 7.1 against it:
   ```java
   public class VersionInfo {
       public static void main(final String[] args) {
           try (KubernetesClient client = new KubernetesClientBuilder().build()) {
               client.getKubernetesVersion();
           }
       }
   }
   ```

_Note: The same seems to happen with 7.2-SNAPSHOT_

### Expected behavior

It returns the `VersionInfo` object with the version information.

### Runtime

Kubernetes (vanilla)

### Kubernetes API Server version

other (please specify in additional context)

### Environment

macOS

### Fabric8 Kubernetes Client Logs

```shell

```

### Additional context

The `/version` endpoint on my Kube 1.33 cluster returns this:

```json
{
  "major": "1",
  "minor": "33",
  "emulationMajor": "1",
  "emulationMinor": "33",
  "minCompatibilityMajor": "1",
  "minCompatibilityMinor": "32",
  "gitVersion": "v1.33.0",
  "gitCommit": "60a317eadfcb839692a68eab88b2096f4d708f4f",
  "gitTreeState": "clean",
  "buildDate": "2025-04-23T13:00:14Z",
  "goVersion": "go1.24.2",
  "compiler": "gc",
  "platform": "linux/amd64"
}
```

The new fields seem to be the cause of the issue.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,7 +13,8 @@
 * Fix #6930: Add support for Boolean enums in the java-generator
 * Fix #6941: HasMetadata.getApiVersion no slash when empty group
 * Fix #6982: (java-generator) Double default field values with `d` suffix
-* FIX #6987: Kube API Test startup fails on readiness SSL check
+* Fix #6987: Kube API Test startup fails on readiness SSL check
+* Fix #7037: getKubernetesVersion works in Kubernetes v1.33.0
 
 #### Improvements
 * Fix #6763: (crd-generator) YAML output customization

--- a/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/VersionInfo.java
+++ b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/VersionInfo.java
@@ -15,144 +15,163 @@
  */
 package io.fabric8.kubernetes.client;
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Getter;
+
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.Date;
 
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Getter
 public class VersionInfo {
   public static final class VersionKeys {
     private VersionKeys() {
     }
 
-    public static final String BUILD_DATE = "buildDate";
-    public static final String GIT_COMMIT = "gitCommit";
-    public static final String GIT_VERSION = "gitVersion";
-    public static final String MAJOR = "major";
-    public static final String MINOR = "minor";
-    public static final String GIT_TREE_STATE = "gitTreeState";
-    public static final String PLATFORM = "platform";
-    public static final String GO_VERSION = "goVersion";
-    public static final String COMPILER = "compiler";
     public static final String BUILD_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ssX";
   }
 
-  private Date buildDate;
-  private String gitCommit;
-  private String gitVersion;
   private String major;
   private String minor;
+  private Integer emulationMajor;
+  private Integer emulationMinor;
+  private Integer minCompatibilityMajor;
+  private Integer minCompatibilityMinor;
+  private String gitCommit;
+  private String gitVersion;
   private String gitTreeState;
-  private String platform;
+  private Date buildDate;
   private String goVersion;
   private String compiler;
-
-  public Date getBuildDate() {
-    return buildDate;
-  }
-
-  public String getGitCommit() {
-    return gitCommit;
-  }
-
-  public String getGitVersion() {
-    return gitVersion;
-  }
-
-  public String getMajor() {
-    return major;
-  }
-
-  public String getMinor() {
-    return minor;
-  }
-
-  public String getGitTreeState() {
-    return gitTreeState;
-  }
-
-  public String getPlatform() {
-    return platform;
-  }
-
-  public String getGoVersion() {
-    return goVersion;
-  }
-
-  public String getCompiler() {
-    return compiler;
-  }
+  private String platform;
 
   private VersionInfo() {
   }
 
-  public static class Builder {
-    private VersionInfo versionInfo = new VersionInfo();
+  public static final class Builder {
+    private String major;
+    private String minor;
+    private Integer emulationMajor;
+    private Integer emulationMinor;
+    private Integer minCompatibilityMajor;
+    private Integer minCompatibilityMinor;
+    private String gitCommit;
+    private String gitVersion;
+    private String gitTreeState;
+    private Date buildDate;
+    private String goVersion;
+    private String compiler;
+    private String platform;
 
     public Builder() {
     }
 
     public Builder(VersionInfo versionInfo) {
       if (versionInfo != null) {
-        this.versionInfo.buildDate = versionInfo.getBuildDate();
-        this.versionInfo.gitCommit = versionInfo.getGitCommit();
-        this.versionInfo.gitVersion = versionInfo.getGitVersion();
-        this.versionInfo.major = versionInfo.getMajor();
-        this.versionInfo.minor = versionInfo.getMinor();
-        this.versionInfo.gitTreeState = versionInfo.getGitTreeState();
-        this.versionInfo.platform = versionInfo.getPlatform();
-        this.versionInfo.goVersion = versionInfo.getGoVersion();
-        this.versionInfo.compiler = versionInfo.getCompiler();
+        this.major = versionInfo.getMajor();
+        this.minor = versionInfo.getMinor();
+        this.emulationMajor = versionInfo.getEmulationMajor();
+        this.emulationMinor = versionInfo.getEmulationMinor();
+        this.minCompatibilityMajor = versionInfo.getMinCompatibilityMajor();
+        this.minCompatibilityMinor = versionInfo.getMinCompatibilityMinor();
+        this.gitCommit = versionInfo.getGitCommit();
+        this.gitVersion = versionInfo.getGitVersion();
+        this.gitTreeState = versionInfo.getGitTreeState();
+        this.buildDate = versionInfo.getBuildDate();
+        this.goVersion = versionInfo.getGoVersion();
+        this.compiler = versionInfo.getCompiler();
+        this.platform = versionInfo.getPlatform();
       }
     }
 
     public Builder withBuildDate(String buildDate) throws ParseException {
       if (buildDate != null) {
-        this.versionInfo.buildDate = new SimpleDateFormat(VersionKeys.BUILD_DATE_FORMAT).parse(buildDate);
+        this.buildDate = new SimpleDateFormat(VersionKeys.BUILD_DATE_FORMAT).parse(buildDate);
       }
       return this;
     }
 
-    public Builder withGitCommit(String gitCommit) {
-      this.versionInfo.gitCommit = gitCommit;
-      return this;
-    }
-
-    public Builder withGitVersion(String gitVersion) {
-      this.versionInfo.gitVersion = gitVersion;
-      return this;
-    }
-
     public Builder withMajor(String major) {
-      this.versionInfo.major = major;
+      this.major = major;
       return this;
     }
 
     public Builder withMinor(String minor) {
-      this.versionInfo.minor = minor;
+      this.minor = minor;
+      return this;
+    }
+
+    public Builder withEmulationMajor(Integer emulationMajor) {
+      this.emulationMajor = emulationMajor;
+      return this;
+    }
+
+    public Builder withEmulationMinor(Integer emulationMinor) {
+      this.emulationMinor = emulationMinor;
+      return this;
+    }
+
+    public Builder withMinCompatibilityMajor(Integer minCompatibilityMajor) {
+      this.minCompatibilityMajor = minCompatibilityMajor;
+      return this;
+    }
+
+    public Builder withMinCompatibilityMinor(Integer minCompatibilityMinor) {
+      this.minCompatibilityMinor = minCompatibilityMinor;
+      return this;
+    }
+
+    public Builder withGitCommit(String gitCommit) {
+      this.gitCommit = gitCommit;
+      return this;
+    }
+
+    public Builder withGitVersion(String gitVersion) {
+      this.gitVersion = gitVersion;
       return this;
     }
 
     public Builder withGitTreeState(String gitTreeState) {
-      this.versionInfo.gitTreeState = gitTreeState;
+      this.gitTreeState = gitTreeState;
       return this;
     }
 
-    public Builder withPlatform(String platform) {
-      this.versionInfo.platform = platform;
+    public Builder withBuildDate(Date buildDate) {
+      this.buildDate = buildDate;
       return this;
     }
 
     public Builder withGoVersion(String goVersion) {
-      this.versionInfo.goVersion = goVersion;
+      this.goVersion = goVersion;
       return this;
     }
 
     public Builder withCompiler(String compiler) {
-      this.versionInfo.compiler = compiler;
+      this.compiler = compiler;
+      return this;
+    }
+
+    public Builder withPlatform(String platform) {
+      this.platform = platform;
       return this;
     }
 
     public VersionInfo build() {
+      VersionInfo versionInfo = new VersionInfo();
+      versionInfo.compiler = this.compiler;
+      versionInfo.gitCommit = this.gitCommit;
+      versionInfo.minCompatibilityMajor = this.minCompatibilityMajor;
+      versionInfo.goVersion = this.goVersion;
+      versionInfo.platform = this.platform;
+      versionInfo.gitVersion = this.gitVersion;
+      versionInfo.major = this.major;
+      versionInfo.emulationMajor = this.emulationMajor;
+      versionInfo.emulationMinor = this.emulationMinor;
+      versionInfo.minor = this.minor;
+      versionInfo.minCompatibilityMinor = this.minCompatibilityMinor;
+      versionInfo.gitTreeState = this.gitTreeState;
+      versionInfo.buildDate = this.buildDate;
       return versionInfo;
     }
   }

--- a/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/MixedCrudTest.java
+++ b/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/MixedCrudTest.java
@@ -18,7 +18,6 @@ package io.fabric8.kubernetes.client.mock;
 import io.fabric8.kubernetes.api.model.Pod;
 import io.fabric8.kubernetes.api.model.PodBuilder;
 import io.fabric8.kubernetes.client.KubernetesClient;
-import io.fabric8.kubernetes.client.KubernetesClientException;
 import io.fabric8.kubernetes.client.VersionInfo;
 import io.fabric8.kubernetes.client.server.mock.KubernetesMixedDispatcher;
 import io.fabric8.kubernetes.client.server.mock.KubernetesMockServer;
@@ -37,7 +36,6 @@ import java.util.Map;
 import java.util.Queue;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.junit.jupiter.api.Assertions.assertThrows;
 
 class MixedCrudTest {
 
@@ -56,15 +54,6 @@ class MixedCrudTest {
   void tearDown() {
     client.close();
     server.destroy();
-  }
-
-  @Test
-  @DisplayName("client.getKubernetesVersion, with no expectations, should throw Exception")
-  void versionWithNoExpectationsShouldFail() {
-    // When
-    final RuntimeException exception = assertThrows(RuntimeException.class, client::getKubernetesVersion);
-    // Then
-    assertThat(exception).isNotNull().isInstanceOf(KubernetesClientException.class);
   }
 
   @Test

--- a/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/VersionInfoTest.java
+++ b/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/VersionInfoTest.java
@@ -46,16 +46,24 @@ class VersionInfoTest {
         "    \"gitCommit\": \"e6301f88a8\"," +
         "    \"gitVersion\": \"v1.6.1+5115d708d7\"," +
         "    \"major\": \"3\"," +
-        "    \"minor\": \"6\"" +
+        "    \"minor\": \"6\"," +
+        "    \"emulationMajor\": \"1\"," +
+        "    \"emulationMinor\": \"33\"," +
+        "    \"minCompatibilityMajor\": \"1\"," +
+        "    \"minCompatibilityMinor\": \"32\"" +
         "}").always();
 
-    assertEquals("v1.6.1+5115d708d7", client.getVersion().getGitVersion());
-    assertEquals("e6301f88a8", client.getVersion().getGitCommit());
-    assertEquals("3", client.getVersion().getMajor());
-    assertEquals("6", client.getVersion().getMinor());
-    assertEquals(118, client.getVersion().getBuildDate().getYear());
+    assertEquals("v1.6.1+5115d708d7", client.getKubernetesVersion().getGitVersion());
+    assertEquals("e6301f88a8", client.getKubernetesVersion().getGitCommit());
+    assertEquals("3", client.getKubernetesVersion().getMajor());
+    assertEquals("6", client.getKubernetesVersion().getMinor());
+    assertEquals(1, client.getKubernetesVersion().getEmulationMajor());
+    assertEquals(33, client.getKubernetesVersion().getEmulationMinor());
+    assertEquals(1, client.getKubernetesVersion().getMinCompatibilityMajor());
+    assertEquals(32, client.getKubernetesVersion().getMinCompatibilityMinor());
+    assertEquals(118, client.getKubernetesVersion().getBuildDate().getYear());
     assertEquals(new SimpleDateFormat(VersionInfo.VersionKeys.BUILD_DATE_FORMAT).parse("2018-03-01T14:27:17Z").getTime(),
-        client.getVersion().getBuildDate().getTime());
+        client.getKubernetesVersion().getBuildDate().getTime());
   }
 
   @Test

